lib/SILOptimizer/Utils/InstOptUtils.cpp

```diff
@@ -701,41 +701,58 @@ SILValue swift::
 getConcreteValueOfExistentialBox(AllocExistentialBoxInst *existentialBox,
                                   SILInstruction *ignoreUser) {
   StoreInst *singleStore = nullptr;
-  for (Operand *use : getNonDebugUses(existentialBox)) {
+  SmallVector<Operand *, 32> worklist;
+  SmallPtrSet<Operand *, 8> addedToWorklistPreviously;
+  for (auto *use : getNonDebugUses(existentialBox)) {
+    worklist.push_back(use);
+    addedToWorklistPreviously.insert(use);
+  }
+
+  while (!worklist.empty()) {
+    auto *use = worklist.pop_back_val();
     SILInstruction *user = use->getUser();
     switch (user->getKind()) {
-      case SILInstructionKind::StrongRetainInst:
-      case SILInstructionKind::StrongReleaseInst:
-        break;
-      case SILInstructionKind::ProjectExistentialBoxInst: {
-        auto *projectedAddr = cast<ProjectExistentialBoxInst>(user);
-        for (Operand *addrUse : getNonDebugUses(projectedAddr)) {
-          if (auto *store = dyn_cast<StoreInst>(addrUse->getUser())) {
-            assert(store->getSrc() != projectedAddr &&
-                   "cannot store an address");
-            // Bail if there are multiple stores.
-            if (singleStore)
-              return SILValue();
-            singleStore = store;
-            continue;
-          }
-          // If there are other users to the box value address then bail out.
-          return SILValue();
+    case SILInstructionKind::StrongRetainInst:
+    case SILInstructionKind::StrongReleaseInst:
+    case SILInstructionKind::DestroyValueInst:
+    case SILInstructionKind::EndBorrowInst:
+      break;
+    case SILInstructionKind::CopyValueInst:
+    case SILInstructionKind::BeginBorrowInst:
+      // Look through copy_value, begin_borrow
+      for (SILValue result : user->getResults())
+        for (auto *transitiveUse : result->getUses())
+          if (!addedToWorklistPreviously.insert(transitiveUse).second)
+            worklist.push_back(use);
```

Comment:
Was re-reading through this code fixing what you were talking about @atrick and found this thinko. I think this should be transitiveUse and I typed.

Reply:
oops

Reply:
yea... I'm not happy about it. But at least we caught it early!